modules/node-shared/src/main/scala/org/tessellation/node/shared/http/routes/SnapshotRoutes.scala

```diff
@@ -40,78 +43,102 @@ final case class SnapshotRoutes[F[_]: Async: Hasher: KryoSerializer, S <: Snapsh
   implicit def binaryAndJsonEncoders[A <: AnyRef: Encoder]: List[EntityEncoder[F, A]] =
     List(BinaryCodec.encoder[F, A], CirceEntityEncoder.circeEntityEncoder[F, A])
 
+  private def badRequestNodeNotReady(currentState: NodeState): F[Response[F]] =
+    BadRequest(("message" ->> s"Node should be in Ready state before fetching. Current state: $currentState") :: HNil)
```

Comment:
Not sure BadRequest is correct - it's supposed to mean a client error and node not being ready yet is not a client error.

Reply:
I would also not expose the actual state. I would just say that node or api is not ready.

Reply:
I support that /\. `4xx` status code is reserved for client error, but this is a server error

Reply:
What status do you suggest, I know that should be something different than 200

Reply:
I think that `503 Service Unavailable` would fit that purpose

Reply:
Got it, I agree

Reply:
@IPadawans Here is the documentation for `4xx` vs `5xx`

>[6.5](https://www.rfc-editor.org/rfc/rfc7231#section-6.5).  Client Error 4xx
>   The 4xx (Client Error) class of status code indicates that the client
   seems to have erred.  Except when responding to a HEAD request, the
   server SHOULD send a representation containing an explanation of the
   error situation, and whether it is a temporary or permanent
   condition.  These status codes are applicable to any request method.
   User agents SHOULD display any included representation to the user.

> [6.6](https://www.rfc-editor.org/rfc/rfc7231#section-6.6).  Server Error 5xx
>   The 5xx (Server Error) class of status code indicates that the server
   is aware that it has erred or is incapable of performing the
   requested method.  Except when responding to a HEAD request, the
   server SHOULD send a representation containing an explanation of the
   error situation, and whether it is a temporary or permanent